core/common/src/main/java/alluxio/grpc/GrpcManagedChannelPool.java

```diff
@@ -86,32 +146,21 @@ public ManagedChannel acquireManagedChannel(ChannelKey channelKey) {
    * @param channelKey host address
    */
   public void releaseManagedChannel(ChannelKey channelKey) {
-
     boolean disposeChannel = false;
     try (LockResource lockShared = new LockResource(mLock.readLock())) {
       if (mChannels.containsKey(channelKey) && mChannels.get(channelKey).dereference()) {
         disposeChannel = true;
       }
     }
-
     if (disposeChannel) {
       try (LockResource lockExclusive = new LockResource(mLock.writeLock())) {
         if (mChannels.containsKey(channelKey)) {
           mChannels.get(channelKey).reference();
           if (mChannels.get(channelKey).dereference()) {
-            ManagedChannel channel = mChannels.remove(channelKey).reference();
-            channel.shutdown();
-            try {
-              channel.awaitTermination(
-                  Configuration.getMs(PropertyKey.MASTER_GRPC_CHANNEL_SHUTDOWN_TIMEOUT),
-                  TimeUnit.MILLISECONDS);
-            } catch (InterruptedException e) {
-              Thread.currentThread().interrupt();
-              // Allow thread to exit.
-            } finally {
-              channel.shutdownNow();
-            }
-            Verify.verify(channel.isShutdown());
+            // Mark channel for destruction.
```

Comment:
What was the issue with destroying here (or in other words, how does async destruction help things)?

Reply:
Just an optimization. This was not the main issue. 
Delaying the actual channel destruction helps for concurrent clients to grab and activate inactive channels. Otherwise even in a highly concurrent scenario it'd be fairly often for the channel to get destroyed and recreated.